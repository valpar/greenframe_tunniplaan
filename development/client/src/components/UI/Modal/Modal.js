import { Fragment, useEffect } from "react";
import ReactDOM from "react-dom";

const Backdrop = (props) => {
  useEffect(() => {
    const themeColorMeta = document.querySelector('meta[name="theme-color"]');

    themeColorMeta.setAttribute("content", "#5C5C5C");

    return () => {
      themeColorMeta.setAttribute("content", "#fff");
    };
  }, []);

  return (
    <div
      data-testid="backdrop"
      className="fixed top-0 left-0 w-full h-screen z-20 bg-neutral-900 opacity-70"
      onClick={props.onClose}
    />
  );
};

const ModalOverlay = (props) => {
  return (
    <div
      data-testid="modal-overlay"
      className={`${props.onHidden ? "hidden" : ""} fixed top-1/3 w-11/12 ${
        props.onCenter ? "" : "lg:ml-32"
      } lg:w-auto h-auto left-1/2 -translate-x-1/2 bg-white rounded z-30 shadow-lg animate-slideDown lg:top-1/4`}
    >
      <div
        className={`absolute ${
          props.modalOverlay ? "flex" : "hidden"
        } justify-center items-center w-full h-full bg-neutral-900 opacity-70 z-30`}
      ></div>
      <div className="p-4">
        <div>{props.children}</div>
      </div>
    </div>
  );
};
const portalElement = document.getElementById("overlays");

const Modal = (props) => {
  return (
    <Fragment>
      {ReactDOM.createPortal(
        <Backdrop onClose={props.onClose} />,
        portalElement
      )}
      {ReactDOM.createPortal(
        <ModalOverlay
          onCenter={props.onCenter}
          onHidden={props.onHidden}
          modalOverlay={props.overlay}
        >
          {props.children}
        </ModalOverlay>,
        portalElement
      )}
    </Fragment>
  );
};

export default Modal;
