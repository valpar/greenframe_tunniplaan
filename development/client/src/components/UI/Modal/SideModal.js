import { useState } from "react";
import { Fragment, useEffect } from "react";
import ReactDOM from "react-dom";

const Backdrop = (props) => {
  useEffect(() => {
    const themeColorMeta = document.querySelector('meta[name="theme-color"]');

    themeColorMeta.setAttribute("content", "#5C5C5C");

    return () => {
      themeColorMeta.setAttribute("content", "#fff");
    };
  }, []);

  return (
    <div
      className="fixed top-0 left-0 w-full h-screen z-20 bg-neutral-900 opacity-70"
      onClick={props.onClose}
    />
  );
};

const ModalOverlay = (props) => {
  return (
    <div
      data-testid="modal-overlay"
      className={`${
        props.onHidden ? "animate-slideRight" : "animate-slideRightOff"
      } fixed top-20 w-full h-full left-1/2 -translate-x-1/2 bg-white z-30 shadow-lg`}
    >
      <div
        className={`absolute ${
          props.modalOverlay ? "flex" : "hidden"
        } justify-center items-center w-full h-full bg-neutral-900 opacity-70 z-30`}
      ></div>
      <div className="p-4">
        <div>{props.children}</div>
      </div>
    </div>
  );
};
const portalElement = document.getElementById("overlays");

const SideModal = (props) => {
  return (
    <Fragment>
      {ReactDOM.createPortal(
        <ModalOverlay onHidden={props.onHidden} modalOverlay={props.overlay}>
          {props.children}
        </ModalOverlay>,
        portalElement
      )}
    </Fragment>
  );
};

export default SideModal;
