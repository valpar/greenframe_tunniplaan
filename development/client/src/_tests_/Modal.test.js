import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Modal from '../components/UI/Modal/Modal';

beforeAll(() => {
    const meta = document.createElement('meta');
    meta.setAttribute('name', 'theme-color');
    document.head.appendChild(meta);
  });
  
afterAll(() => {
  // eslint-disable-next-line testing-library/no-node-access
  document.head.removeChild(document.querySelector('meta[name="theme-color"]'));
});

// kui seda ei tee (createPortal), siis target "is not DOM element"
jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: (element, target) => element, 
}));

describe('Modal', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<Modal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with Backdrop and ModalOverlay', () => {
    render(<Modal />);
    
    expect(screen.getByTestId('backdrop')).toBeInTheDocument();
    expect(screen.getByTestId('modal-overlay')).toBeInTheDocument();
  });
  
});