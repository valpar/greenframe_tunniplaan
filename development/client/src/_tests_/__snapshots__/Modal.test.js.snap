// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
Array [
  <div
    className="fixed top-0 left-0 w-full h-screen z-20 bg-neutral-900 opacity-70"
    data-testid="backdrop"
  />,
  <div
    className=" fixed top-1/3 w-11/12 lg:ml-32 lg:w-auto h-auto left-1/2 -translate-x-1/2 bg-white rounded z-30 shadow-lg animate-slideDown lg:top-1/4"
    data-testid="modal-overlay"
  >
    <div
      className="absolute hidden justify-center items-center w-full h-full bg-neutral-900 opacity-70 z-30"
    />
    <div
      className="p-4"
    >
      <div />
    </div>
  </div>,
]
`;
